src/main/java/net/minecraftforge/network/simple/SimpleChannel.java

```diff
@@ -181,11 +181,34 @@ private static <MSG> MessageBuilder<MSG> forType(final SimpleChannel channel, fi
             return builder;
         }
 
+        /**
+         * Set the message encoder, which writes this message to a {@link FriendlyByteBuf}.
+         *
+         * The encoder is called <em>immediately</em> {@link #send(PacketDistributor.PacketTarget, Object) when the
+         * packet is sent}. This means encoding typically occurs on the main server/client thread rather than on the
+         * network thread.
+         *
+         * However, this behaviour should not be relied on, and the encoder should try to be thread-safe and not
+         * interact with the current game state.
+         *
+         * @param encoder The message encoder.
+         * @return This message builder, for chaining.
+         */
         public MessageBuilder<MSG> encoder(BiConsumer<MSG, FriendlyByteBuf> encoder) {
             this.encoder = encoder;
             return this;
         }
 
+        /**
+         * Set the message decoder, which reads the message from a {@link FriendlyByteBuf}.
+         *
+         * The decoder is called when the message is received on the network thread. The decoder should not attempt to
+         * access or mutate any game state, deferring that until the {@link #consumer(ToBooleanBiFunction) the message
+         * is handled}.
```

Comment:
Paragraphs other than the first should either be separated by or surrounded with `<p>` tags. (I personally prefer the latter.) Additionally, `{@linkplain}` should be used when the label would normally not be in monospace.

```suggestion
         * <p>The decoder is called when the message is received on the network thread. The decoder should not attempt to
         * access or mutate any game state, deferring that until the {@linkplain #consumer(ToBooleanBiFunction) the message
         * is handled}.</p>
```